frontend/src/components/forms/formUtils.ts

```diff
@@ -0,0 +1,68 @@
+import { InputHTMLAttributes } from 'react'
+import { RegisterOptions } from 'react-hook-form'
+
+/**
+ * Adds error messages for the following validations:
+ * - required
+ * - min
+ * - max
+ * - minLength
+ * - maxLength
+ */
+export const augmentRegisterOptionsForInput = (
+  registerOptions: RegisterOptions,
+  inputProps: InputHTMLAttributes<HTMLInputElement>,
+) => {
+  const options = Object.assign({}, registerOptions)
+
+  if (inputProps.type === 'number') {
+    options.valueAsNumber = true
+
+    // A required input with type="number" will always fail if the value is 0
+    // Therefore, we need a custom validation
+    if (options.required === true) {
+      delete options.required
+      options.validate = Object.assign({}, options.validate, {
+        required: (value: any) => {
+          if (value == null || isNaN(value) || value === '') {
+            return 'This field is required'
```

Comment:
Added bonus: defining error messages in this file will make localization a little easier.